src/libs/nav2_tasks/include/nav2_tasks/task_client.hpp

```diff
@@ -70,6 +70,24 @@ class TaskClient
     cancelPub_->publish(msg);
   }
 
+  bool waitForServer(std::chrono::milliseconds timeout = std::chrono::milliseconds(3000))
```

Comment:
I think the best choice for default behavior is to wait forever. That's how API functions typically behave if no timeout is specified.